src/transformers/modeling_tf_bart.py

```diff
@@ -1130,15 +1199,17 @@ def adjust_logits_during_generation(self, logits, cur_len, max_length):
         return logits
 
     @staticmethod
-    def _force_token_id_to_be_generated(scores, token_id) -> None:
+    def _force_token_id_to_be_generated(scores, token_id, inverted=False) -> None:
         """force one of token_ids to be generated by setting prob of all other tokens to 0 (logprob=-float("inf"))"""
+        # TODO: https://github.com/huggingface/transformers/issues/7954
         output_list = []
-
         # Is there a better way to do scores[:, [x for if x != token_id]] = -float("inf") in TF?
         bs, vocab_size = scores.shape
```

Comment:
We should not use `.shape` in TF, but the `shape_list()` function to not run into problems for compilation.